### PR TITLE
Fix build for old GMP version

### PR DESCRIPTION
**Pull request by 4tXJ7f**
Older versions of GMP in combination with newer versions of GCC and
C++11 cause errors [0]. This commit adds the necessary includes of
<cstddef>.

[0] https://gcc.gnu.org/gcc-4.9/porting_to.html

### CODE REVIEW

**Comment by 4tXJ7f** [comment]
The issue was originally reported by Radu Iosif via e-mail. He confirmed that this fix helps.

File: src/util/gmp_util.h [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #ifndef __CVC4__GMP_UTIL_H
 #define __CVC4__GMP_UTIL_H
 
+#include <cstddef>
```
Comment (timothy-king) [inline src/util/gmp_util.h]: Add comments for why these are needed.

File: src/util/rational_gmp_imp.h [hunk 2]
```diff
@@ -20,6 +20,7 @@
 #ifndef __CVC4__RATIONAL_H
 #define __CVC4__RATIONAL_H
 
+#include <cstddef>
```
Comment (timothy-king) [inline src/util/rational_gmp_imp.h]: ditto

**Comment by 4tXJ7f** [comment]
Added a comment as requested.

**Comment by aniemetz** [comment]
LGTM

**Pull request merged**